### PR TITLE
Comment out hard coded opensearch.yml

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
This PR stops feeding in a hard coded opensearch.yml into the pods. We are expecting to generate this output from running the install demo configuration script from the security plugin side.
 
### Issues Resolved
[List any issues this PR will resolve. You should likely open an issue if one does not already exist.]
 
### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: charts/opensearch/values.yaml [hunk 1]
```diff
@@ -31,71 +31,71 @@ global:
...
-    # Bind to all interfaces because we don't know what IP address Docker will assign to us.
-    network.host: 0.0.0.0
-
-    # Setting network.host to a non-loopback address enables the annoying bootstrap checks. "Single-node" mode disables them again.
-    # Implicitly done if ".singleNode" is set to "true".
-    # discovery.type: single-node
-
-    # Start OpenSearch Security Demo Configuration
-    # WARNING: revise all the lines below before you go into production
-    plugins:
```
Comment (derek-ho) [inline charts/opensearch/values.yaml:60]: This file should be generated via the demo configuration script, if it is hard coded we will need to maintain this in two places.

**Review by TheAlgo — COMMENTED** [review]
But this sample yaml is very helpful for anyone starting opensearch right?

**Comment by smlx** [comment]
I don't understand this change. This is a helm chart, they are configured via values files.. how are you generating your configuration otherwise?

**Comment by DarshitChanpura** [comment]
@smlx @TheAlgo When an OpenSearch image is a downloaded and cluster is spun up, the Security plugin runs install demo configuration by default as part of install. During this phase, opensearch.yml is loaded with security related values. Hence we do not need to hardcode opensearch.yml. Feel free to experiment and lmk if otherwise.

File: charts/opensearch/CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 ### Security
 
+---
+## [2.17.3]
```
Comment (peterzhuamazon) [inline charts/opensearch/CHANGELOG.md:17]: Use 2.18.1 as I put #503 as #2.18.0.

File: charts/opensearch/Chart.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 2.17.2
+version: 2.17.3
```
Comment (peterzhuamazon) [inline charts/opensearch/Chart.yaml:18]: 2.18.1

File: charts/opensearch/CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+### Added
+### Changed
+### Deprecated
+### Removed
+- Removed hard coded opensearch.yml
```
Comment (peterzhuamazon) [inline charts/opensearch/CHANGELOG.md:22]: This is a breaking change, please be clear in the changelog.

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]

**Review by peterzhuamazon — APPROVED** [review]

**Comment by DarshitChanpura** [comment]
@prudhvigodithi Could you please review this?

**Comment by DarshitChanpura** [comment]
@prudhvigodithi Could you please add 1 more review?

**Comment by prudhvigodithi** [comment]
Hey Just tested this and works fine with removing the `.Values.config` , basically when user adds the `opensearch.yml` through `.Values.config`, it gets added to `/usr/share/opensearch/config/opensearch.yml` overriding the default. Also one advantage is next if there is any change in  `opensearch.yml` and added through `.Values.config`, the pods get auto restarted and loads the new values from `opensearch.yml`.  So @DarshitChanpura and @derek-ho any specific reason we are removing, we can leave it as well right ?

**Comment by DarshitChanpura** [comment]
> So @DarshitChanpura and @derek-ho any specific reason we are removing, we can leave it as well right ?

From what I understand, the opensearch.yml always gets created when demo configuration script is run (which is default). If in case the user decides to pass in custom configuration, they can always pass it in via .values.yml. In either of these two cases, the hardcoded opensearch.yml serves no purpose. Which is why we are removing this? @derek-ho Please add if I missed anything here.

**Comment by smlx** [comment]
I don't know if it is on the roadmap of the Opensearch image to avoid generating the demo config at runtime but IMO it should be. Generating config at runtime has the following consequences for docker and k8s:

* it slows down container startup
* root filesystem can't be read only
* can't run the container as an arbitrary user (has to have permissions to write the config file)

IMO it makes more sense to aim for immutable images with defined injection points for configuration rather than generating config at runtime. Therefore I think this PR is a move in the wrong direction.

**Comment by prudhvigodithi** [comment]
Instead of commenting out `opensearch.yml`, I have raised a PR https://github.com/opensearch-project/helm-charts/pull/516, that address configMap Read-only file system error when passed via `.Values.config`, with this the user can either use the default `opensearch.yml` or pass the `opensearch.yml` via `.Values.config`. The file will be created as `-rw-r--r-- 1 opensearch opensearch 3646 Feb  7 03:22 opensearch.yml`, having RW access within the cluster,  please check.

@TheAlgo @DarshitChanpura @peterzhuamazon @bbarani @derek-ho @smlx

**Comment by derek-ho** [comment]
I can close this out, since I tested out the alternative also works and it seems the direction of the community is not towards removing this.

**Comment by DarshitChanpura** [comment]
This can be closed out since the changes in #516 addresses the read-only issue when writing demo configuration.

**Pull request closed without merging**